### PR TITLE
Add some more tests after PoC phase :)

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

File: spec/Denormalizer/CompositeDenormalizerSpec.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        $this->shouldImplement(DenormalizerInterface::class);
+    }
+
+    function it_does_not_support_a_message_if_there_are_no_denormalizers()
+    {
+        $amqpMessage = new AMQPMessage('Message body');
+
+        $this->supports($amqpMessage)->shouldReturn(false);
+        $this->shouldThrow(DenormalizationFailedException::class)->during('denormalize', [$amqpMessage]);
```
Comment (Arminek) [inline spec/Denormalizer/CompositeDenormalizerSpec.php:25]: Are you sure that it should throw `DenormalizationFailedException` when there are no denormalizers registered?

File: spec/Denormalizer/CompositeDenormalizerSpec.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        $firstDenormalizer->supports($amqpMessage)->willReturn(false);
+        $firstDenormalizer->denormalize($amqpMessage)->shouldNotBeCalled();
+        $secondDenormalizer->supports($amqpMessage)->willReturn(false);
+        $secondDenormalizer->denormalize($amqpMessage)->shouldNotBeCalled();;
+
+        $this->addDenormalizer($firstDenormalizer);
+        $this->addDenormalizer($secondDenormalizer);
+
+        $this->supports($amqpMessage)->shouldReturn(false);
+        $this->shouldThrow(DenormalizationFailedException::class)->during('denormalize', [$amqpMessage]);
```
Comment (Arminek) [inline spec/Denormalizer/CompositeDenormalizerSpec.php:62]: This behaviour can cuz some problems. How about something like when I cannot denormalize message I don't care about it?

File: spec/Denormalizer/CompositeDenormalizerSpec.php (hunk 1 above)
Comment (pamil) [reply spec/Denormalizer/CompositeDenormalizerSpec.php:25]: Yeah, if there are no denormalizers, you can't denormalize any message.

File: spec/Denormalizer/CompositeDenormalizerSpec.php (hunk 2 above)
Comment (pamil) [reply spec/Denormalizer/CompositeDenormalizerSpec.php:62]: Right now the default consumer will log exception messages and ignore not supported messages.

Comment (Arminek) [reply spec/Denormalizer/CompositeDenormalizerSpec.php:62]: 👍

**Review by Arminek — APPROVED** [review]
👍

**Pull request merged**